src/components/react-json-schemaForm/GeometryField/PointField/PointField.js

```diff
@@ -10,46 +10,62 @@ const Label = ({ name, required }) => (
 );
 
 const PointField = ({
-  idSchema: { coordinates: { $ids } },
+  idSchema: { coordinates: { $id } },
   formData,
+  formData: { coordinates },
   onChange,
   t,
   schema: {
     required = [],
   },
   disabled,
 }) => {
-  const [coords, setCoords] = useState(formData.coordinates);
   const isRequired = required.includes('coordinates');
 
+  const refLatitude = useRef(null);
+  const refLongitude = useRef(null);
 
-  const handleChange = index => ({ target: { value } }) => {
-    const coordinates = [...coords];
-    coordinates[index] = Number(value);
-    setCoords(coordinates);
+  useEffect(() => {
+    const [latitude = '', longitude = ''] = coordinates;
+    refLatitude.current.value = latitude;
+    refLongitude.current.value = longitude;
+    onChange(formData);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [coordinates]);
+
+
+  const handleChange = () => {
     onChange({ ...formData, coordinates });
   };
 
   return (
     <div>
-      {['latitude', 'longitude'].map((name, index) => (
-        <FormGroup
-          key={name}
-          label={(
-            <Label name={t(`jsonSchema.geometryField.${name}`)} required={isRequired} />
+      {['latitude', 'longitude'].map((name, index) => {
+        const ref = !index ? refLatitude : refLongitude;
+        const { current } = ref;
+        const { value = '' } = current || {};
+        return (
+          <FormGroup
+            key={name}
+            label={(
+              <Label name={t(`jsonSchema.geometryField.${name}`)} required={isRequired} />
           )}
-          labelFor={`${$ids}-${index}`}
-        >
-          <InputGroup
-            id={`${$ids}-${index}`}
-            disabled={disabled}
-            defaultValue={coords[index]}
-            onChange={handleChange(index)}
-            required={isRequired}
-            type="number"
-          />
-        </FormGroup>
-      ))}
+            labelFor={`${$id}-${index}`}
+          >
+            <InputGroup
+              id={`${$id}-${index}`}
+              disabled={disabled}
+              defaultValue={value}
+              inputMode="numeric"
+              inputRef={ref}
+              onChange={handleChange}
+              pattern="[+-]?([0-9]*[.])?[0-9]+"
+              required={isRequired}
+              type="text"
+            />
+          </FormGroup>
+        );
+      })}
```

Comment:
Using a ref for each fields seems to be here to allow the `[].map()`.
You could avoid this quite useless ref by just using a simpler intermediate component :

```jsx
const CustomCoordField = ({ name, id, ...props }) => {
  const { required } = props;

  return (
          <FormGroup
            label={<Label name={t(`jsonSchema.geometryField.${name}`)} required={required} />}
            labelFor={`${$id}-${name}`}
          >
            <InputGroup
              id={`${$id}-${name}`}
              inputMode="numeric"
              pattern="[+-]?([0-9]*[.])?[0-9]+"
              type="text"
              {...props}
            />
          </FormGroup>
  );
};

// ...

const PointField = ({
  // ...
}) => {
  // ...
  return (
    <div>
      <CustomCoordField
        name="latitude"
        onChange={handleChange}
        required={isRequired}
        defaultValue={coordinates[0]}
      />
      <CustomCoordField
        name="longitude"
        onChange={handleChange}
        required={isRequired}
        defaultValue={coordinates[1]}
      />
    </div>
  );
};
```

Reply:
This is not the case.
If the user draw the point in the map, i use ref to set coordinates on each field (cf the useEffect at the top)

```js
  useEffect(() => {
    const [latitude = '', longitude = ''] = coordinates;
    refLatitude.current.value = latitude;
    refLongitude.current.value = longitude;
    onChange(formData);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [coordinates]);
```

Reply:
But i check to refacto this

Reply:
I still do not see the point of using refs this way: if you have to manage the value of inputs, using value prop of each one and corresponding states is the classical way.
And if you have multiple states having to be changed together, `useReducer()` is the best solution.
I probably miss something else...